### PR TITLE
Modulemd.ModuleIndexMerger.test_merger_with_real_world_data into C

### PR DESCRIPTION
**Pull request by MR-M1M3**
Copied python test to C for https://github.com/fedora-modularity/libmodulemd/blob/master/modulemd/tests/ModulemdTests/merger.py#L238-254

### CODE REVIEW

File: modulemd/tests/test-modulemd-merger.c [hunk 1]
```diff
@@ -273,3 +273,31 @@ main (int argc, char *argv[])
 
   return g_test_run ();
 }
+
+test_merger_with_real_world_data (ModuleStreamFixture *fixture,
```
Comment (OrionStar25) [inline modulemd/tests/test-modulemd-merger.c]: Every function in C needs a return type unlike in Python. Here the return type would be `static void`.

Comment (MR-M1M3) [reply modulemd/tests/test-modulemd-merger.c]: I've updated the file again and fixed some errors.

Comment (MR-M1M3) [reply modulemd/tests/test-modulemd-merger.c]: Here are the errors.
https://paste.centos.org/view/bc4bbcc9

Comment (OrionStar25) [reply modulemd/tests/test-modulemd-merger.c]: Please push your changes to this PR so that I can see them.

File: modulemd/modulemd-util.c [hunk 2]
```diff
@@ -360,7 +360,7 @@ modulemd_validate_nevra (const gchar *nevra)
         {
           break;
         }
-      else if (*i == '-')
+      if (*i == '-')
```
Comment (OrionStar25) [inline modulemd/modulemd-util.c]: Why this?

Comment (sgallagher) [reply modulemd/modulemd-util.c]: The code formatter that we run during `ninja test` recently started to drop unneeded `else` statements. It's unrelated to this change. I've pushed this to the current master branch so it doesn't keep showing up.

**Comment by OrionStar25** [comment]
@MR-M1M3 Please rebase and run `ninja test` before committing.

**Comment by MR-M1M3** [comment]
I ran ninja test before commiting the changes, still the error exists

**Pull request merged**